src/map/conquest_data.h

```diff
@@ -0,0 +1,68 @@
+#ifndef SERVER_CONQUEST_DATA_H
+#define SERVER_CONQUEST_DATA_H
+
+#include <string>
+
+#include "common/cbasetypes.h"
+#include "conquest_zmq.h"
+#include "zone.h"
+
+class ConquestData
+{
+public:
+    ConquestData();
+
+    /**
+     * Loads the latest conquest data from DB.
+     * Should only ever be done on map initialization.
+     * World server should be updating conquest data periodically.
+     */
+    void load(std::unique_ptr<SqlConnection>& sql);
+
+    /**
+     * Gets the influence points for a given nation in a given region.
+     */
+    int32 getInfluence(REGION_TYPE region, uint8 nation);
+
+    /**
+     * Gets the owner of a given region. That is, the one controlling this region since last tally.
+     */
+    uint8 getRegionOwner(REGION_TYPE region);
+
+    /**
+     * Gets the number of regions controlled by a given nation.
+     */
+    uint8 getRegionControlCount(uint8 nation);
+
+    /**
+     * Gets the number of regions controlled by a given nation in the
+     * previous tally.
+     */
+    uint8 getPrevRegionControlCount(uint8 nation);
+
+    /**
+     * Gets the array of region controls, indexed by regionId.
+     */
+    auto getRegionControls() -> std::vector<region_control_t>;
+
+    /**
+     * Adds the given influence points to the given nation for the given
+     * region.
+     */
+    void addInfluencePoints(int points, uint8 nation, REGION_TYPE region);
+    /**
+     * Updates the influence points to match those given.
+     */
+    void updateInfluencePoints(std::vector<influence_t> influencePoints);
```

Comment:
If we're copying in influencePoints, presumably we're also not modifying it. So we could change this to be a const& and save ourselves a copy

Reply:
Done